pymchelper/executor/options.py

```diff
@@ -0,0 +1,134 @@
+
+
+class SH12AEnvironmentLinux:
+    """
+    TODO
+    """
+    executable_file = 'shieldhit'
+
+
+class SH12AEnvironmentWindows:
+    """
+    TODO
+    """
+    executable_file = 'shieldhit.exe'
+
+
+class MCOptions:
+    """
+    TODO
+    """
+    def __init__(self, input_path, executable_path=None, user_opt=None):
+        self.input_path = input_path
+        self._mc_environment = self._discover_mc_engine()
+        self.executable_path = executable_path
+        if self.executable_path is None:
+            self.executable_path = self._discover_mc_executable()
+        self.user_opt = user_opt if user_opt else ''
+        if self.user_opt:
+            self._validate_user_opt(self.user_opt)
+        self.workspace = '.'
+
+    def set_rng_seed(self, rng_seed):
+        """
+        TODO this method is specific to SH12A, more general should be added
+        """
+        options_list = self.user_opt.split()
+        if '-N' not in options_list:
+            self.user_opt += " -N {:d}".format(rng_seed)
+        else:
+            location = options_list.index('-N')
+            options_list[location + 1] = str(rng_seed)
+            self.user_opt = ' '.join(options_list)
+
+    def set_no_of_primaries(self, number_of_primaries):
+        """
+        TODO this method is specific to SH12A, more general should be added
+        """
+        options_list = self.user_opt.split()
+        if '-n' not in options_list:
+            self.user_opt += " -n {:d}".format(number_of_primaries)
+        else:
+            location = options_list.index('-n')
+            options_list[location + 1] = str(number_of_primaries)
+            self.user_opt = ' '.join(options_list)
+
+    @staticmethod
+    def _validate_user_opt(user_opt):
+        """
+        TODO this method is specific to SH12A, more general should be added
+        """
+        options_list = user_opt.split()
+        options_set = set(options_list)
+        unsupported = {'-b', '--beamfile', '-g', '--geofile', '-m', '--matfile', '-d', '--detectfile'}
+        if options_set & unsupported:
+            raise SyntaxError("Unsupported option encountered: {:s}".format(",".join(options_set & unsupported)))
+        if len(options_list) > 1:
+            last_item = options_list[-1]
+            before_last_item = options_list[-2]
+            single_options = {'-h', '--help', '-V', '--version', '-v', '--verbose', '-s', '--silent',
+                              '-l', '--legacy-bdo'}
+            if not last_item.startswith('-'):
+                if before_last_item in single_options:
+                    raise SyntaxError("Seems like workspace: {:s}".format(last_item))
+                if len(options_list) > 2 and options_list[-3].startswith('-'):
+                    raise SyntaxError("Seems like workspace: {:s}".format(last_item))
+        if len(options_list) == 1 and not options_list[0].startswith('-'):
+            raise SyntaxError("Seems like workspace: {:s}".format(options_list[0]))
```

Comment:
Whole function needs to be explained

Reply:
I will write down more comments, thanks for remark !